src/model/account/Account.ts

```diff
@@ -137,9 +137,13 @@ export class Account {
      * @param data - Data to be signed
      * @return {string} - Signed data result
      */
-    public signData(data: string): string {
-        return convert.uint8ToHex(KeyPair.sign(this.keyPair,
-                            convert.hexToUint8(convert.utf8ToHex(data)),
-                        ));
+    public signData(data: string | Uint8Array): string {
```

Comment:
in case we want to support all kind of `Uint8Array` and so on, TypeScript is common to use the `|` operator to support more than one type. But in this case, I would have, as you suggested, two different methods.

`signData` for strings, `signBinary` for Uint8Array for example?

Reply:
sorry for the late reply on this,
so what is the conclusion exactly? should have two functions?
my initial idea was inspired by cryptojs hash functions where they accept multiple data types like in this example:
```typescript
MD5(message: string | LibWordArray, key?: string | WordArray, ...options: any[]): WordArray;
```

Reply:
Two functions better. Yes, I know some libraries use multiple arguments types to do overloading, but with two functions seems more clear.